code/game/turfs/open.dm

```diff
@@ -97,19 +97,39 @@
 			wet_overlay = null
 		var/turf/open/floor/F = src
 		if(istype(F))
-			wet_overlay = image('icons/effects/water.dmi', src, "wet_floor_static")
+			if(wet_setting == TURF_WET_ICE)
+				wet_overlay = image('icons/turf/overlays.dmi', src, "snowfloor")
+			else
+				wet_overlay = image('icons/effects/water.dmi', src, "wet_floor_static")
 		else
 			wet_overlay = image('icons/effects/water.dmi', src, "wet_static")
 		overlays += wet_overlay
 
-	spawn(rand(790, 820)) // Purely so for visual effect
-		if(!istype(src, /turf)) //Because turfs don't get deleted, they change, adapt, transform, evolve and deform. they are one and they are all.
-			return
-		MakeDry(wet_setting)
+	if(wet < TURF_WET_ICE)
+		addtimer(src, "MakeDry", rand(790, 820))
 
 /turf/open/proc/MakeDry(wet_setting = TURF_WET_WATER)
-	if(wet > wet_setting)
+	if(!istype(src, /turf)) //Because turfs don't get deleted, they change, adapt, transform, evolve and deform. they are one and they are all.
+		return
```

Comment:
I was thinking the same thing when i saw it. But i didnt add that, i just moved it.